File: app/models/tournament.rb
class Tournament < ApplicationRecord
  validates :name, presence: true
  validates :date, presence: true
  validates :per_team, presence: true
  validates :entry_code, presence: true

  has_many :holes
  has_many :scores, through: :holes
  has_many :teams
  has_many :players, through: :teams

end

File: client/src/components/holes/Tournament.js
import React from 'react'
import Jumbotron from 'react-bootstrap/Jumbotron'

const Tournament = props => {
  return (
    <div>
      <Jumbotron>
        <h1> {props.tournament.name} </h1>
        <p> Date: {props.tournament.date} </p>
        <p> Players per team: {props.tournament.per_team} </p>
        <p> Entry Code: {props.tournament.entry_code} </p>
      </Jumbotron>
    </div>
  )

}

export default Tournament
